neuro_san/internals/chat/data_driven_chat_session.py

```diff
@@ -240,3 +243,24 @@ def prepare_chat_context(self, chat_message_history: List[Dict[str, Any]]) -> Di
         }
 
         return chat_context
+
+    def create_outgoing_message_processor(self) -> MessageProcessor:
+        """
+        :return: A MessageProcessor that filters messages outgoing to the client.
+                How this works is based on settings on the front man.
+                Can be None.
+        """
```

Comment:
Not entirely happy about the location here, but it gets things going.

Basically look to the front man as to whether or not we should be attempting to look for structure in any and all messages going back to the client and if so, set up a MessageProcessor that will tweak all the messages for external consumption in the DirectAgentSessions.

This might move when some consolidation between sync/async DirectAgentSession happens.